app/models/jwt_token.rb

```diff
@@ -0,0 +1,63 @@
+require 'jwt'
+require 'base64'
+
+class JwtToken < Struct.new(:token)
+  class << self
+    def encode(user, secret)
+      payload = prepare_payload(user, secret)
+      new JWT.encode(payload, secret)
+    end
+
+    private
+
+    def prepare_payload(user, secret)
+      jti_raw = [secret, iat].join(':')
+      jti = Digest::MD5.hexdigest(jti_raw)
+      { user_id: user.id, iat: iat, jti: jti }
+    end
+
+    def iat
+      Time.now.to_i
+    end
+  end
+
+  def decode
+    return nil if token.blank?
```

Comment:
why do you return here `nil`, and on the usage you are converting it into empty hash?